### PR TITLE
Keep parentheses of `SequenceExpression` for `ExportDefaultDeclaration`

### PR DESCRIPTION
**Pull request by fisker**
Fixes: #7490

Can't find other cases need cover

- [x] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [x] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/printer-estree.js [hunk 1]
```diff
@@ -1612,8 +1612,13 @@ function printPathNoParens(path, options, print, args) {
         }, "expressions");
         return group(concat(parts));
       }
+      const needParentheses = parent.type === "ExportDefaultDeclaration";
       return group(
-        concat([join(concat([",", line]), path.map(print, "expressions"))])
+        concat([
+          needParentheses ? "(" : "",
+          join(concat([",", line]), path.map(print, "expressions")),
+          needParentheses ? ")" : ""
```
Comment (thorn0) [inline src/language-js/printer-estree.js]: It's not the right way to fix it. Please modify `case "SequenceExpression"` in `needs-parens.js` insteed.

Comment (fisker) [reply src/language-js/printer-estree.js]: Thanks changed. how about now?

**Comment by fisker** [comment]
That function returns earlier, maybe an if ?

**Comment by thorn0** [comment]
It also now returns `true` in some nested scenarios where previously it didn't. Did you investigate them?

**Comment by thorn0** [comment]
"returns earlier" is not important. There is the big `switch` and this logic should live in the corresponding `case`.

**Comment by fisker** [comment]
When `parent.type === "ExportDefaultDeclaration"`, result is decided by `shouldWrapFunctionForExportDefault` didn't go through `case "SequenceExpression"`.

And `case "SequenceExpression"` defaults to `true` already.

**Comment by fisker** [comment]
Is current code OK? I'm going to sleep.

**Comment by sosukesuzuki** [comment]
Although different from the purpose of this PR, I think all `SequenceExpressions` should be parenthesized. They are hard to read...

**Pull request merged**